data_steward/analytics/cdr_ops/rt_cdr_qc/cdr_deid_qa_report7_cope_survey.py

```diff
@@ -290,127 +286,137 @@
-                ignore_index = True) 
+    summary = summary.append({'query' : 'Query6 The concepts are not suppressed in observation table' , 'result' : 'Failure'},
+                             ignore_index = True)
 
-# # 7 done Vaccine-related concepts as these EHR-submitted COVID concepts are allowed from RT 
+# # 7 done Vaccine-related concepts as these EHR-submitted COVID concepts are allowed from RT
 # DC-2374
-# this query was from DC-1752 
+# this query was from DC-1752
+#
+# Aslo, ensure concepts existing in combined are maintained in deid, and concepts absent in combine should not appear in deid.  See [DC-3631](https://precisionmedicineinitiative.atlassian.net/browse/DC-3631)
 
 query = JINJA_ENV.from_string("""
-
-DECLARE vocabulary_tables DEFAULT ['vocabulary', 'concept', 'source_to_concept_map', 
+DECLARE vocabulary_tables DEFAULT ['vocabulary', 'concept', 'source_to_concept_map',
                                    'concept_class', 'concept_synonym', 'concept_ancestor',
                                    'concept_relationship', 'relationship', 'drug_strength'];
-SELECT table_name,column_name
 
+SELECT table_name,column_name
 FROM `{{project_id}}.{{deid_cdr}}.INFORMATION_SCHEMA.COLUMNS` c
 JOIN `{{project_id}}.{{deid_cdr}}.__TABLES__` t
  ON c.table_name = t.table_id
-WHERE 
-     table_name NOT IN UNNEST(vocabulary_tables) and 
+WHERE
+     table_name NOT IN UNNEST(vocabulary_tables) and
   t.row_count > 0
   AND table_name NOT LIKE '\\\_%'
   AND table_name in ('procedure_occurrence','drug_exposure')
   AND column_name in ('procedure_concept_id','procedure_source_concept_id','drug_concept_id','drug_source_concept_id')
 """)
-q = query.render(project_id=project_id,deid_cdr=deid_cdr)
-target_tables=execute(client, q)
+q = query.render(project_id=project_id,
+                 deid_cdr=deid_cdr)
+target_tables = execute(client, q)
 target_tables.shape
 
+target_tables
+
 
 # +
 #table_name="drug_exposure"
 #@column_name="drug_concept_id"
 
-def my_sql(table_name,column_name):
+def target_of(table_name, column_name):
 
     query = JINJA_ENV.from_string("""
-    
-SELECT 
+SELECT
 '{{table_name}}' AS table_name,
 '{{column_name}}' AS column_name,
+concept_id_in_combined,
 COUNT(*) AS row_counts,
-CASE WHEN 
-  COUNT(*) > 0
+CASE WHEN
+  COUNT(*) > 0 AND sub.concept_id_in_combined IS NOT NULL
   THEN 0 ELSE 1
 END
  AS Failure_row_counts
- 
-FROM `{{project_id}}.{{deid_cdr}}.{{table_name}}` c
-JOIN  `{{project_id}}.{{deid_cdr}}.concept` on concept_id={{column_name}}
- WHERE (
-        -- done by name and vocab -- -- this alone should be enough, no need for others --
-        REGEXP_CONTAINS(concept_name, r'(?i)(COVID)') AND
-        REGEXP_CONTAINS(concept_name, r'(?i)(VAC)') AND 
-        vocabulary_id not in ('PPI')
-    ) OR (
-        -- done by code  and vocab --
+FROM `{{project_id}}.{{deid_cdr}}.procedure_occurrence` c
```

Comment:
Here it should not be `procedure_occurrence` it should be `{{table_name}}`. The table_name in the ticket is added for using it as example and testing. if you are looping through tables and columns for in your case this should be templated to as all other ones.